src/mo_types/type.ml

```diff
@@ -14,9 +14,10 @@ type obj_sort =
  | Module
  | Memory          (* (codegen only): stable memory serialization format *)
 
+type stable_sort = Flexible | Stable
 type async_sort = Fut | Cmp
 type shared_sort = Query | Write | Composite
-type 'a shared = Local | Shared of 'a
+type 'a shared = Local of stable_sort | Shared of 'a
```

Comment:
```suggestion
type ('a, 'b) shared = Local | Stable of 'b | Shared of 'a
```
I wonder if this might have reduced some code changes elsewhere... Don't do it now, just a note to self.
Then you could attach the free variable sets to stable functions only, using a suitable instantiations of 'b.

Reply:
Can indeed be a simplification. Worth checking